### ISSUE TITLE
Enabling LTO causes spurious rebuilds

### ISSUE THREAD

**Issue by xobs** [rust-lang/cargo#8750]
**Problem**
With LTO set to `thin` or `fat`, doing a `cargo build --target riscv32imac-unknown-none-elf` followed by `cargo run` followed by `cargo build --target riscv32imac-unknown-none-elf` causes Cargo to rebuild some packages even when nothing has changed.

Setting `lto="off"` prevents this behavior.

**Steps**

1. Ensure you have the `riscv32imac-unknown-none-elf` toolchain
2. Clone the `xous-core` repository
3. Build an example such as `shell`
4. Run a program
5. Rebuild `shell`

This series of steps should demonstrate the problem:
```
rustup target add riscv32imac-unknown-none-elf
git clone https://github.com/betrusted-io/xous-core.git
cd xous-core
git checkout 6cb1aad93ca908bb394b9e7c13274b9f55d5aa8b
cargo build --release --target riscv32imac-unknown-none-elf -p shell
cargo run --release -p tools --bin copy-object
cargo build --release --target riscv32imac-unknown-none-elf -p shell # This will cause a partial rebuild
cargo run --release -p tools --bin copy-object # This will also cause a partial rebuild
```

**Notes**

If you modify `Cargo.toml` to remove `lto="thin"`, or change it to `lto="off`", then the issue goes away.

Output of `cargo version`: cargo 1.46.0 (149022b1d 2020-07-17)

This may be due to the fact that `cargo run` and `cargo build` are targeting different architectures.

**Comment by xobs** [comment]
As suggested in #8337, I can run `cargo build` with `CARGO_LOG=cargo::core::compiler::fingerprint=trace`. It naturally gives a lot of spew:

```
xobs@ci:~/foo/xous-core$ CARGO_LOG=cargo::core::compiler::fingerprint=trace cargo build --release --target riscv32imac-unknown-none-elf -p shell
[2020-10-04T07:28:02Z DEBUG cargo::core::compiler::fingerprint] fingerprint at: /home/xobs/foo/xous-core/target/riscv32imac-unknown-none-elf/release/.fingerprint/shell-35a055e8e1884d92/bin-shell
[2020-10-04T07:28:02Z DEBUG cargo::core::compiler::fingerprint] old local fingerprints deps "/home/xobs/.cargo/registry/src/github.com-1ecc6299db9ec823/bitflags-1.2.1" precalculated="1.2.1"
[2020-10-04T07:28:02Z DEBUG cargo::core::compiler::fingerprint] max output mtime for "/home/xobs/.cargo/registry/src/github.com-1ecc6299db9ec823/bitflags-1.2.1" is "/home/xobs/foo/xous-core/target/release/build/bitflags-beaeafc5be58f280/build_script_build-beaeafc5be58f280" 1601796273.473621608s
[2020-10-04T07:28:02Z DEBUG cargo::core::compiler::fingerprint] all paths up-to-date relative to "/home/xobs/foo/xous-core/target/release/.fingerprint/bitflags-beaeafc5be58f280/dep-build-script-build-script-build" mtime=1601796273.097623715s
[2020-10-04T07:28:02Z DEBUG cargo::core::compiler::fingerprint] filesystem up-to-date "/home/xobs/.cargo/registry/src/github.com-1ecc6299db9ec823/bitflags-1.2.1"
[2020-10-04T07:28:02Z DEBUG cargo::core::compiler::fingerprint] max output mtime for "/home/xobs/.cargo/registry/src/github.com-1ecc6299db9ec823/bitflags-1.2.1" is "/home/xobs/foo/xous-core/target/riscv32imac-unknown-none-elf/release/build/bitflags-711410ae194d8896/output" 1601796273.477621585s
[2020-10-04T07:28:02Z DEBUG cargo::core::compiler::fingerprint] max dep mtime for "/home/xobs/.cargo/registry/src/github.com-1ecc6299db9ec823/bitflags-1.2.1" is "/home/xobs/foo/xous-core/target/release/build/bitflags-beaeafc5be58f280/build_script_build-beaeafc5be58f280" 1601796273.473621608s
[2020-10-04T07:28:02Z DEBUG cargo::core::compiler::fingerprint] filesystem up-to-date "/home/xobs/.cargo/registry/src/github.com-1ecc6299db9ec823/bitflags-1.2.1"
[2020-10-04T07:28:02Z DEBUG cargo::core::compiler::fingerprint] max output mtime for "/home/xobs/.cargo/registry/src/github.com-1ecc6299db9ec823/bitflags-1.2.1" is "/home/xobs/foo/xous-core/target/riscv32imac-unknown-none-elf/release/deps/libbitflags-d97608ce41bd2533.rlib" 1601796273.517621361s
[2020-10-04T07:28:02Z DEBUG cargo::core::compiler::fingerprint] max dep mtime for "/home/xobs/.cargo/registry/src/github.com-1ecc6299db9ec823/bitflags-1.2.1" is "/home/xobs/foo/xous-core/target/riscv32imac-unknown-none-elf/release/build/bitflags-711410ae194d8896/output" 1601796273.477621585s
[2020-10-04T07:28:02Z DEBUG cargo::core::compiler::fingerprint] all paths up-to-date relative to "/home/xobs/foo/xous-core/target/riscv32imac-unknown-none-elf/release/.fingerprint/bitflags-d97608ce41bd2533/dep-lib-bitflags" mtime=1601796273.489621518s
[2020-10-04T07:28:02Z DEBUG cargo::core::compiler::fingerprint] filesystem up-to-date "/home/xobs/.cargo/registry/src/github.com-1ecc6299db9ec823/bitflags-1.2.1"
[2020-10-04T07:28:02Z DEBUG cargo::core::compiler::fingerprint] old local fingerprints deps "/home/xobs/.cargo/registry/src/github.com-1ecc6299db9ec823/byteorder-1.3.4" precalculated="1.3.4"
[2020-10-04T07:28:02Z DEBUG cargo::core::compiler::fingerprint] max output mtime for "/home/xobs/.cargo/registry/src/github.com-1ecc6299db9ec823/byteorder-1.3.4" is "/home/xobs/foo/xous-core/target/release/build/byteorder-0de1c3d3e49642d6/build_script_build-0de1c3d3e49642d6" 1601796273.741620105s
[2020-10-04T07:28:02Z DEBUG cargo::core::compiler::fingerprint] all paths up-to-date relative to "/home/xobs/foo/xous-core/target/release/.fingerprint/byteorder-0de1c3d3e49642d6/dep-build-script-build-script-build" mtime=1601796273.097623715s
[2020-10-04T07:28:02Z DEBUG cargo::core::compiler::fingerprint] filesystem up-to-date "/home/xobs/.cargo/registry/src/github.com-1ecc6299db9ec823/byteorder-1.3.4"
[2020-10-04T07:28:02Z DEBUG cargo::core::compiler::fingerprint] max output mtime for "/home/xobs/.cargo/registry/src/github.com-1ecc6299db9ec823/byteorder-1.3.4" is "/home/xobs/foo/xous-core/target/riscv32imac-unknown-none-elf/release/build/byteorder-186c4dafd57a8088/output" 1601796273.745620083s
[2020-10-04T07:28:02Z DEBUG cargo::core::compiler::fingerprint] max dep mtime for "/home/xobs/.cargo/registry/src/github.com-1ecc6299db9ec823/byteorder-1.3.4" is "/home/xobs/foo/xous-core/target/release/build/byteorder-0de1c3d3e49642d6/build_script_build-0de1c3d3e49642d6" 1601796273.741620105s
[2020-10-04T07:28:02Z DEBUG cargo::core::compiler::fingerprint] filesystem up-to-date "/home/xobs/.cargo/registry/src/github.com-1ecc6299db9ec823/byteorder-1.3.4"
[2020-10-04T07:28:02Z DEBUG cargo::core::compiler::fingerprint] max output mtime for "/home/xobs/.cargo/registry/src/github.com-1ecc6299db9ec823/byteorder-1.3.4" is "/home/xobs/foo/xous-core/target/riscv32imac-unknown-none-elf/release/deps/libbyteorder-36c736c2160550d3.rlib" 1601796273.897619232s
[2020-10-04T07:28:02Z DEBUG cargo::core::compiler::fingerprint] max dep mtime for "/home/xobs/.cargo/registry/src/github.com-1ecc6299db9ec823/byteorder-1.3.4" is "/home/xobs/foo/xous-core/target/riscv32imac-unknown-none-elf/release/build/byteorder-186c4dafd57a8088/output" 1601796273.745620083s
[2020-10-04T07:28:02Z DEBUG cargo::core::compiler::fingerprint] all paths up-to-date relative to "/home/xobs/foo/xous-core/target/riscv32imac-unknown-none-elf/release/.fingerprint/byteorder-36c736c2160550d3/dep-lib-byteorder" mtime=1601796273.753620038s
[2020-10-04T07:28:02Z DEBUG cargo::core::compiler::fingerprint] filesystem up-to-date "/home/xobs/.cargo/registry/src/github.com-1ecc6299db9ec823/byteorder-1.3.4"
[2020-10-04T07:28:02Z DEBUG cargo::core::compiler::fingerprint] max output mtime for "/home/xobs/.cargo/registry/src/github.com-1ecc6299db9ec823/embedded-graphics-0.6.2" is "/home/xobs/foo/xous-core/target/riscv32imac-unknown-none-elf/release/deps/libembedded_graphics-00d14c44027e403c.rlib" 1601796274.917613514s
[2020-10-04T07:28:02Z DEBUG cargo::core::compiler::fingerprint] max dep mtime for "/home/xobs/.cargo/registry/src/github.com-1ecc6299db9ec823/embedded-graphics-0.6.2" is "/home/xobs/foo/xous-core/target/riscv32imac-unknown-none-elf/release/deps/libbyteorder-36c736c2160550d3.rmeta" 1601796273.893619254s
[2020-10-04T07:28:02Z DEBUG cargo::core::compiler::fingerprint] all paths up-to-date relative to "/home/xobs/foo/xous-core/target/riscv32imac-unknown-none-elf/release/.fingerprint/embedded-graphics-00d14c44027e403c/dep-lib-embedded-graphics" mtime=1601796273.893619254s
[2020-10-04T07:28:02Z DEBUG cargo::core::compiler::fingerprint] filesystem up-to-date "/home/xobs/.cargo/registry/src/github.com-1ecc6299db9ec823/embedded-graphics-0.6.2"
[2020-10-04T07:28:02Z DEBUG cargo::core::compiler::fingerprint] new local fingerprints deps "/home/xobs/foo/xous-core/xous-rs"
[2020-10-04T07:28:02Z DEBUG cargo::core::compiler::fingerprint] max output mtime for "/home/xobs/foo/xous-core/xous-rs" is "/home/xobs/foo/xous-core/target/release/build/xous-b6769b5e793ce494/build_script_build-b6769b5e793ce494" 1601796273.345622325s
[2020-10-04T07:28:02Z DEBUG cargo::core::compiler::fingerprint] all paths up-to-date relative to "/home/xobs/foo/xous-core/target/release/.fingerprint/xous-b6769b5e793ce494/dep-build-script-build-script-build" mtime=1601796273.097623715s
[2020-10-04T07:28:02Z DEBUG cargo::core::compiler::fingerprint] filesystem up-to-date "/home/xobs/foo/xous-core/xous-rs"
[2020-10-04T07:28:02Z DEBUG cargo::core::compiler::fingerprint] max output mtime for "/home/xobs/foo/xous-core/xous-rs" is "/home/xobs/foo/xous-core/target/riscv32imac-unknown-none-elf/release/build/xous-9742fdf1e5122a7b/output" 1601796273.349622302s
[2020-10-04T07:28:02Z DEBUG cargo::core::compiler::fingerprint] max dep mtime for "/home/xobs/foo/xous-core/xous-rs" is "/home/xobs/foo/xous-core/target/release/build/xous-b6769b5e793ce494/build_script_build-b6769b5e793ce494" 1601796273.345622325s
[2020-10-04T07:28:02Z DEBUG cargo::core::compiler::fingerprint] all paths up-to-date relative to "/home/xobs/foo/xous-core/target/riscv32imac-unknown-none-elf/release/build/xous-9742fdf1e5122a7b/output" mtime=1601796273.349622302s
[2020-10-04T07:28:02Z DEBUG cargo::core::compiler::fingerprint] filesystem up-to-date "/home/xobs/foo/xous-core/xous-rs"
[2020-10-04T07:28:02Z DEBUG cargo::core::compiler::fingerprint] new local fingerprints deps "/home/xobs/.cargo/registry/src/github.com-1ecc6299db9ec823/proc-macro2-1.0.19"
[2020-10-04T07:28:02Z DEBUG cargo::core::compiler::fingerprint] max output mtime for "/home/xobs/.cargo/registry/src/github.com-1ecc6299db9ec823/proc-macro2-1.0.19" is "/home/xobs/foo/xous-core/target/release/build/proc-macro2-ccdfc0b9e33cc25b/build_script_build-ccdfc0b9e33cc25b" 1601796273.597620913s
[2020-10-04T07:28:02Z DEBUG cargo::core::compiler::fingerprint] all paths up-to-date relative to "/home/xobs/foo/xous-core/target/release/.fingerprint/proc-macro2-ccdfc0b9e33cc25b/dep-build-script-build-script-build" mtime=1601796273.093623738s
[2020-10-04T07:28:02Z DEBUG cargo::core::compiler::fingerprint] filesystem up-to-date "/home/xobs/.cargo/registry/src/github.com-1ecc6299db9ec823/proc-macro2-1.0.19"
[2020-10-04T07:28:02Z DEBUG cargo::core::compiler::fingerprint] max output mtime for "/home/xobs/.cargo/registry/src/github.com-1ecc6299db9ec823/proc-macro2-1.0.19" is "/home/xobs/foo/xous-core/target/release/build/proc-macro2-8ec4a53ca09bb64e/output" 1601796273.605620867s
[2020-10-04T07:28:02Z DEBUG cargo::core::compiler::fingerprint] max dep mtime for "/home/xobs/.cargo/registry/src/github.com-1ecc6299db9ec823/proc-macro2-1.0.19" is "/home/xobs/foo/xous-core/target/release/build/proc-macro2-ccdfc0b9e33cc25b/build_script_build-ccdfc0b9e33cc25b" 1601796273.597620913s
[2020-10-04T07:28:02Z DEBUG cargo::core::compiler::fingerprint] all paths up-to-date relative to "/home/xobs/foo/xous-core/target/release/build/proc-macro2-8ec4a53ca09bb64e/output" mtime=1601796273.605620867s
[2020-10-04T07:28:02Z DEBUG cargo::core::compiler::fingerprint] filesystem up-to-date "/home/xobs/.cargo/registry/src/github.com-1ecc6299db9ec823/proc-macro2-1.0.19"
[2020-10-04T07:28:02Z DEBUG cargo::core::compiler::fingerprint] max output mtime for "/home/xobs/.cargo/registry/src/github.com-1ecc6299db9ec823/unicode-xid-0.2.1" is "/home/xobs/foo/xous-core/target/release/deps/libunicode_xid-f442c63735117183.rlib" 1601796273.185623222s
[2020-10-04T07:28:02Z DEBUG cargo::core::compiler::fingerprint] all paths up-to-date relative to "/home/xobs/foo/xous-core/target/release/.fingerprint/unicode-xid-f442c63735117183/dep-lib-unicode-xid" mtime=1601796273.093623738s
[2020-10-04T07:28:02Z DEBUG cargo::core::compiler::fingerprint] filesystem up-to-date "/home/xobs/.cargo/registry/src/github.com-1ecc6299db9ec823/unicode-xid-0.2.1"
[2020-10-04T07:28:02Z DEBUG cargo::core::compiler::fingerprint] max output mtime for "/home/xobs/.cargo/registry/src/github.com-1ecc6299db9ec823/proc-macro2-1.0.19" is "/home/xobs/foo/xous-core/target/release/deps/libproc_macro2-e375f0cf392d168f.rlib" 1601796276.193606362s
[2020-10-04T07:28:02Z DEBUG cargo::core::compiler::fingerprint] max dep mtime for "/home/xobs/.cargo/registry/src/github.com-1ecc6299db9ec823/proc-macro2-1.0.19" is "/home/xobs/foo/xous-core/target/release/build/proc-macro2-8ec4a53ca09bb64e/output" 1601796273.605620867s
[2020-10-04T07:28:02Z DEBUG cargo::core::compiler::fingerprint] max dep mtime for "/home/xobs/.cargo/registry/src/github.com-1ecc6299db9ec823/proc-macro2-1.0.19" is "/home/xobs/foo/xous-core/target/release/deps/libunicode_xid-f442c63735117183.rmeta" 1601796273.153623401s
[2020-10-04T07:28:02Z DEBUG cargo::core::compiler::fingerprint] all paths up-to-date relative to "/home/xobs/foo/xous-core/target/release/.fingerprint/proc-macro2-e375f0cf392d168f/dep-lib-proc-macro2" mtime=1601796273.617620801s
[2020-10-04T07:28:02Z DEBUG cargo::core::compiler::fingerprint] filesystem up-to-date "/home/xobs/.cargo/registry/src/github.com-1ecc6299db9ec823/proc-macro2-1.0.19"
[2020-10-04T07:28:02Z DEBUG cargo::core::compiler::fingerprint] max output mtime for "/home/xobs/.cargo/registry/src/github.com-1ecc6299db9ec823/quote-1.0.7" is "/home/xobs/foo/xous-core/target/release/deps/libquote-53c7a5bed7fdb5f7.rlib" 1601796274.833613985s
[2020-10-04T07:28:02Z DEBUG cargo::core::compiler::fingerprint] max dep mtime for "/home/xobs/.cargo/registry/src/github.com-1ecc6299db9ec823/quote-1.0.7" is "/home/xobs/foo/xous-core/target/release/deps/libproc_macro2-e375f0cf392d168f.rmeta" 1601796273.857619455s
[2020-10-04T07:28:02Z DEBUG cargo::core::compiler::fingerprint] all paths up-to-date relative to "/home/xobs/foo/xous-core/target/release/.fingerprint/quote-53c7a5bed7fdb5f7/dep-lib-quote" mtime=1601796273.857619455s
[2020-10-04T07:28:02Z DEBUG cargo::core::compiler::fingerprint] filesystem up-to-date "/home/xobs/.cargo/registry/src/github.com-1ecc6299db9ec823/quote-1.0.7"
[2020-10-04T07:28:02Z DEBUG cargo::core::compiler::fingerprint] max output mtime for "/home/xobs/.cargo/registry/src/github.com-1ecc6299db9ec823/cfg-if-0.1.10" is "/home/xobs/foo/xous-core/target/release/deps/libcfg_if-40a30409d4cf2c96.rmeta" 1601796355.181163587s
[2020-10-04T07:28:02Z DEBUG cargo::core::compiler::fingerprint] all paths up-to-date relative to "/home/xobs/foo/xous-core/target/release/.fingerprint/cfg-if-40a30409d4cf2c96/dep-lib-cfg-if" mtime=1601796355.133163856s
[2020-10-04T07:28:02Z DEBUG cargo::core::compiler::fingerprint] filesystem up-to-date "/home/xobs/.cargo/registry/src/github.com-1ecc6299db9ec823/cfg-if-0.1.10"
[2020-10-04T07:28:02Z DEBUG cargo::core::compiler::fingerprint] old local fingerprints deps "/home/xobs/.cargo/registry/src/github.com-1ecc6299db9ec823/getrandom-0.1.14" precalculated="0.1.14"
[2020-10-04T07:28:02Z DEBUG cargo::core::compiler::fingerprint] max output mtime for "/home/xobs/.cargo/registry/src/github.com-1ecc6299db9ec823/getrandom-0.1.14" is "/home/xobs/foo/xous-core/target/release/build/getrandom-4bbc32d185b27d90/build_script_build-4bbc32d185b27d90" 1601796273.329622414s
[2020-10-04T07:28:02Z DEBUG cargo::core::compiler::fingerprint] all paths up-to-date relative to "/home/xobs/foo/xous-core/target/release/.fingerprint/getrandom-4bbc32d185b27d90/dep-build-script-build-script-build" mtime=1601796273.093623738s
[2020-10-04T07:28:02Z DEBUG cargo::core::compiler::fingerprint] filesystem up-to-date "/home/xobs/.cargo/registry/src/github.com-1ecc6299db9ec823/getrandom-0.1.14"
[2020-10-04T07:28:02Z DEBUG cargo::core::compiler::fingerprint] max output mtime for "/home/xobs/.cargo/registry/src/github.com-1ecc6299db9ec823/getrandom-0.1.14" is "/home/xobs/foo/xous-core/target/release/build/getrandom-c285fea7b3fb284f/output" 1601796273.333622392s
[2020-10-04T07:28:02Z DEBUG cargo::core::compiler::fingerprint] max dep mtime for "/home/xobs/.cargo/registry/src/github.com-1ecc6299db9ec823/getrandom-0.1.14" is "/home/xobs/foo/xous-core/target/release/build/getrandom-4bbc32d185b27d90/build_script_build-4bbc32d185b27d90" 1601796273.329622414s
[2020-10-04T07:28:02Z DEBUG cargo::core::compiler::fingerprint] filesystem up-to-date "/home/xobs/.cargo/registry/src/github.com-1ecc6299db9ec823/getrandom-0.1.14"
[2020-10-04T07:28:02Z DEBUG cargo::core::compiler::fingerprint] old local fingerprints deps "/home/xobs/.cargo/registry/src/github.com-1ecc6299db9ec823/libc-0.2.74" precalculated="0.2.74"
[2020-10-04T07:28:02Z DEBUG cargo::core::compiler::fingerprint] max output mtime for "/home/xobs/.cargo/registry/src/github.com-1ecc6299db9ec823/libc-0.2.74" is "/home/xobs/foo/xous-core/target/release/build/libc-18a10c75e6a447af/build_script_build-18a10c75e6a447af" 1601796273.585620980s
[2020-10-04T07:28:02Z DEBUG cargo::core::compiler::fingerprint] all paths up-to-date relative to "/home/xobs/foo/xous-core/target/release/.fingerprint/libc-18a10c75e6a447af/dep-build-script-build-script-build" mtime=1601796273.093623738s
[2020-10-04T07:28:02Z DEBUG cargo::core::compiler::fingerprint] filesystem up-to-date "/home/xobs/.cargo/registry/src/github.com-1ecc6299db9ec823/libc-0.2.74"
[2020-10-04T07:28:02Z DEBUG cargo::core::compiler::fingerprint] max output mtime for "/home/xobs/.cargo/registry/src/github.com-1ecc6299db9ec823/libc-0.2.74" is "/home/xobs/foo/xous-core/target/release/build/libc-290252d30b404745/output" 1601796273.589620957s
[2020-10-04T07:28:02Z DEBUG cargo::core::compiler::fingerprint] max dep mtime for "/home/xobs/.cargo/registry/src/github.com-1ecc6299db9ec823/libc-0.2.74" is "/home/xobs/foo/xous-core/target/release/build/libc-18a10c75e6a447af/build_script_build-18a10c75e6a447af" 1601796273.585620980s
[2020-10-04T07:28:02Z DEBUG cargo::core::compiler::fingerprint] filesystem up-to-date "/home/xobs/.cargo/registry/src/github.com-1ecc6299db9ec823/libc-0.2.74"
[2020-10-04T07:28:02Z DEBUG cargo::core::compiler::fingerprint] max output mtime for "/home/xobs/.cargo/registry/src/github.com-1ecc6299db9ec823/libc-0.2.74" is "/home/xobs/foo/xous-core/target/release/deps/liblibc-c4231246f7345c1f.rlib" 1601796275.481610353s
[2020-10-04T07:28:02Z DEBUG cargo::core::compiler::fingerprint] max dep mtime for "/home/xobs/.cargo/registry/src/github.com-1ecc6299db9ec823/libc-0.2.74" is "/home/xobs/foo/xous-core/target/release/build/libc-290252d30b404745/output" 1601796273.589620957s
[2020-10-04T07:28:02Z DEBUG cargo::core::compiler::fingerprint] all paths up-to-date relative to "/home/xobs/foo/xous-core/target/release/.fingerprint/libc-c4231246f7345c1f/dep-lib-libc" mtime=1601796273.601620890s
[2020-10-04T07:28:02Z DEBUG cargo::core::compiler::fingerprint] filesystem up-to-date "/home/xobs/.cargo/registry/src/github.com-1ecc6299db9ec823/libc-0.2.74"
[2020-10-04T07:28:02Z DEBUG cargo::core::compiler::fingerprint] max output mtime for "/home/xobs/.cargo/registry/src/github.com-1ecc6299db9ec823/getrandom-0.1.14" is "/home/xobs/foo/xous-core/target/release/deps/libgetrandom-f978aba086e4982b.rlib" 1601796349.737194107s
[2020-10-04T07:28:02Z DEBUG cargo::core::compiler::fingerprint] max dep mtime for "/home/xobs/.cargo/registry/src/github.com-1ecc6299db9ec823/getrandom-0.1.14" is "/home/xobs/foo/xous-core/target/release/build/getrandom-c285fea7b3fb284f/output" 1601796273.333622392s
[2020-10-04T07:28:02Z DEBUG cargo::core::compiler::fingerprint] max dep mtime for "/home/xobs/.cargo/registry/src/github.com-1ecc6299db9ec823/getrandom-0.1.14" is "/home/xobs/foo/xous-core/target/release/deps/liblibc-c4231246f7345c1f.rmeta" 1601796274.617615196s
[2020-10-04T07:28:02Z DEBUG cargo::core::compiler::fingerprint] max dep mtime for "/home/xobs/.cargo/registry/src/github.com-1ecc6299db9ec823/getrandom-0.1.14" is "/home/xobs/foo/xous-core/target/release/deps/libcfg_if-40a30409d4cf2c96.rmeta" 1601796355.181163587s
[2020-10-04T07:28:02Z INFO  cargo::core::compiler::fingerprint] dependency on `cfg_if` is newer than we are 1601796355.181163587s > 1601796349.737194107s "/home/xobs/.cargo/registry/src/github.com-1ecc6299db9ec823/getrandom-0.1.14"
[2020-10-04T07:28:02Z DEBUG cargo::core::compiler::fingerprint] max output mtime for "/home/xobs/.cargo/registry/src/github.com-1ecc6299db9ec823/ppv-lite86-0.2.8" is "/home/xobs/foo/xous-core/target/release/deps/libppv_lite86-0932788375a412d0.rlib" 1601796273.677620464s
[2020-10-04T07:28:02Z DEBUG cargo::core::compiler::fingerprint] all paths up-to-date relative to "/home/xobs/foo/xous-core/target/release/.fingerprint/ppv-lite86-0932788375a412d0/dep-lib-ppv-lite86" mtime=1601796273.097623715s
```

The interesting line appears to be:

```
2020-10-04T07:28:02Z DEBUG cargo::core::compiler::fingerprint] max dep mtime for "/home/xobs/.cargo/registry/src/github.com-1ecc6299db9ec823/getrandom-0.1.14" is "/home/xobs/foo/xous-core/target/release/deps/libcfg_if-40a30409d4cf2c96.rmeta" 1601796355.181163587s
[2020-10-04T07:28:02Z INFO  cargo::core::compiler::fingerprint] dependency on `cfg_if` is newer than we are 1601796355.181163587s > 1601796349.737194107s "/home/xobs/.cargo/registry/src/github.com-1ecc6299db9ec823/getrandom-0.1.14"
```

But I'm not sure how to address this.

**Comment by ehuss** [comment]
Thanks for the report!  I believe this has already been fixed via #8657.  Can you try with beta (1.47) to confirm?

**Comment by xobs** [comment]
I can confirm that it is fixed in Beta (1.47). Thank you for finding that fix, and it's nice to know what's causing it.
